omnigibson/prims/cloth_prim.py

```diff
@@ -139,58 +165,84 @@ def particle_positions(self):
         s = self.scale
 
         p_local = np.array(self.get_attribute(attr="points"))
+        p_local = p_local[self._keypoint_idx] if keypoints_only else p_local
         p_world = (r @ (p_local * s).T).T + t
 
         return p_world
 
-    @particle_positions.setter
-    def particle_positions(self, pos):
+    def set_particle_positions(self, pos, keypoints_only=False):
         """
-        Set the particle positions for this instancer
+        Set the particle positions of this cloth
 
         Args:
             np.array: (N, 3) numpy array, where each of the N particles' desired positions are expressed in (x,y,z)
                 cartesian coordinates relative to the world frame
+            keypoints_only (bool): If True, will only set the keypoint particle state
         """
-        assert pos.shape[0] == self._n_particles, \
-            f"Got mismatch in particle setting size: {pos.shape[0]}, vs. number of particles {self._n_particles}!"
-
         r = T.quat2mat(self.get_orientation())
         t = self.get_position()
         s = self.scale
-
         p_local = (r.T @ (pos - t).T).T / s
+
+        if keypoints_only:
+            assert pos.shape[0] == len(self._keypoint_idx), \
+                f"Got mismatch in particle keypoint setting size: {pos.shape[0]}, " \
+                f"vs. number of particle keypoints {len(self._keypoint_idx)}!"
+            # Grab the particle positions and overwrite the specific values
+            # We don't use our API to avoid the transformation overhead
+            raw_pos = np.array(self.get_attribute(attr="points"))
+            raw_pos[self._keypoint_idx] = p_local
+            p_local = raw_pos
+        assert pos.shape[0] == self._n_particles, \
+            f"Got mismatch in particle setting size: {pos.shape[0]}, vs. number of particles {self._n_particles}!"
```

Comment:
should we allow setting particle positions for `keypoints_only=True`? Seems a bit strange.

Reply:
fair point -- I can't really think of a use-case, but I wanted to be consistent with our getters / setters sharing the same interface

Reply:
removed